File: algospot/FESTIVAL/FESTIVAL.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

class Solution {

   static void Main(String[] args) {
      //C
      string line = Console.ReadLine().Trim();
      int C       = int.Parse(line);
      List<float> l_r = new List<float>();

      for(int n=0; n<C; ++n)
      {
         //L, N    
         List<int> l_line_2 = GetArrFromLine(Console.ReadLine().Trim());
         int N = l_line_2[0]; // 6
         int L = l_line_2[1]; // 3
         //l_costs
         List<int> l_costs = GetArrFromLine(Console.ReadLine().Trim());

         List<float> l_rr = new List<float>();

         for(int i=0; i<N-L+1; ++i)
            for(int j=i+L; j<N+1; ++j)
            {
               int len = j-i;
               //Console.WriteLine( i.ToString() + " " + len.ToString() );
               List<int> l = l_costs.GetRange(i, len);
               float f = l.Aggregate(0, (acc, x) => acc + x) / (float)l.Count; 
               l_rr.Add(f);
            }
         l_r.Add(l_rr.Min(x => x));
      }

      foreach(float f in l_r)
         Console.WriteLine(f.ToString("F11"));
   }

   static List<int> GetArrFromLine(string line)
   {
      List<int> l = new List<int>();

      string[] words = line.Split(' ');
      foreach(string w in words)
      {
         l.Add(int.Parse(w));   
      }

      return l;
   }



}

File: algospot/FESTIVAL/FESTIVAL.py
import sys
import gc

C = int(raw_input().strip())

l_r = []

for n_case in range(C):
    gc.collect()

    l       = map((lambda x: int(x)), raw_input().strip().split(' '))
    N       = l[0] # 6
    L       = l[1] # 3

    l_costs = map((lambda x: int(x)), raw_input().strip().split(' ')) # [1, 2, 3, 1, 2, 3]
    
    l_rr = []
    for i in range(N - L + 1):
        for j in range(i + L, N + 1): 
            #print(i, j, N - L + 1, i + L, N)
            l = l_costs[i : j]
            #print(i, j)
            l_rr.append(l)
    #print(l_rr)
    l_rr = map( (lambda x: sum(x)/float(len(x))), l_rr)
    l_r.append(min(l_rr))

for n_r in l_r: print("{0:0.11f}".format(n_r))
